team/challenges/systems/worker.tex

```diff
@@ -0,0 +1,268 @@
+  \end{itemize}    
+    
+ \subsection*{Level 2}
+
+ \subsubsection*{API}
+ \begin{itemize} 
+ \item[] Use mTLS instead of basic authentication and verify client certificate. Set up strong set of cipher suites for TLS and good crypto setup for certificates.
+ \end{itemize}
+
+ \subsection*{Level 3}
+    
+  \subsubsection*{API}
+  \begin{itemize} 
+  \item[] Use \href{https://grpc.io}{GRPC} for API and add a streaming output of a running job process.
+  \item[] Add streaming log output of a running job process.
+  \end{itemize}
+
+  \subsubsection*{Client}
+  \begin{itemize} 
+  \item[] Client should be able to stream the logs.
+  \end{itemize}
+  
+  \subsection*{Level 4}
+    
+  \subsubsection*{Resource control}
+  \begin{itemize}
+  \item[] Add limits for amount of CPU, Memory and Disk IO using system calls.
+  \end{itemize}
+
+  \subsection*{Level 5}
+    
+  \subsubsection*{Isolation}
+  \begin{itemize}
+  \item[] Each job process should be given a separate PID, Mount and Networking namespace using system calls.
+  \end{itemize}
+
+\section{Guidance}
+
+\subsection{Interview process}
+
+The interview team joins the slack channel. The team consists of the engineers who will be working with you.
+Ask them bout the engineering culture, work and life balance, or anything else that you would like to learn about Teleport.
+  
+Before writing the actual code, create a brief design document in Google Docs or markdown in Github and share with the team.
+
+This document should consist of key trade-offs and key design approaches. Please avoid writing overly detailed design document. Use this document to make sure the team could provide feedback on your design and demonstrate that you've investigated the problem space.
+
+Split your code submission using pull requests and give the team an opportunity to review the PRs. A good “rule of thumb” to follow is that the final PR submission is a formality adding a small feature set - it means that the team had an opportunity to contribute the feedback during multiple well defined stages of your work.
+
+Our team will do their best to provide a high quality review of the submitted pull requests in a reasonable time frame. You are spending your time on this, we are going to contribute our time too.
+
+After the final submission, the interview team will assemble and vote using +1, -2 anonymous voting system: +1 is submitted whenever a team member accepts the submission, -2 otherwise.
+
+In case of a positive result, we will connect you to our HR team who will collect one-two references and will work out other details. You can start the reference collection process in parallel if you would like to speed up the process.
+
+After reference collection, our ops team will send you an offer.
+
+In case of a negative score result, hiring manager will contact you and send a list of the key observations from the team that affected the result.
```

Comment:
```suggestion
In case of a negative score result, hiring manager will contact you and share a list of the key observations from the team that affected the result.
```